src/main/java/com/flightstats/hub/webhook/strategy/TimedWebhookStrategy.java

```diff
@@ -234,8 +234,19 @@ private DateTime getNextTime() {
                 }
                 return getNextTime.apply(lastTime);
             }
+        };
+    }
 
-        }, getOffsetSeconds.get(), period, TimeUnit.SECONDS);
+    @Override
+    public void start(Webhook webhook, ContentPath initialStartingPath) {
+        ThreadFactory factory = new ThreadFactoryBuilder().setNameFormat(webhook.getBatch() + "-webhook-" + webhook.getName() + "-%s").build();
```

Comment:
I know this isn't your code, but if this webhook is started multiple times the threads will have identical names. Not sure how possible that is.

Reply:
It *shouldn't* happen. I'm not sure how bomb-proof that logic is, but I can check.

Reply:
I moved this up into the constructor (so that it'll increment properly if `start` is called multiple times) and also added 4 characters of randomness in case a `TimedWebhookStrategy` is initialized multiple times for the same webhook. Might be overkill; the 4 characters of randomness is likely enough...